graph/manifest.go

```diff
@@ -108,6 +111,11 @@ func (s *TagStore) newManifest(localName, remoteName, tag string) ([]byte, error
 			}
 
 			checksum = tarSum.Sum(nil)
+
+			// Save checksum value
+			if err := layer.SaveCheckSum(s.graph.ImageRoot(layer.ID), checksum); err != nil {
+				return nil, fmt.Errorf("Error storing image checksum: %s", err)
```

Comment:
`SaveCheckSum` already returns a rich error message. On error, this will output:

    Error storing image checksum: Error storing checksum in foo/bar/checksum: some error message